ui/src/components/executions/Logs.vue

```diff
@@ -119,17 +148,64 @@
                 openedTaskrunsCount: 0,
                 raw_view: false,
                 logIndicesByLevel: Object.fromEntries(LogUtils.levelOrLower(undefined).map(level => [level, []])),
-                logCursor: undefined
+                logCursor: undefined,
+                temporalCursor: undefined,
+                logLineRefs: {},
+                filterUpdateCount: 0
             };
         },
         created() {
             this.level = (this.$route.query.level || localStorage.getItem("defaultLogLevel") || "INFO");
             this.filter = (this.$route.query.q || undefined);
         },
+        watch:{
+            level: {
+                handler() {
+                    if (this.raw_view) {
+                        this.fetchLogs();
+                    }
+                },
+                immediate: true
+            },
+            temporalCursor(newValue) {
+                if (newValue !== undefined && this.raw_view) {
+                    this.scrollToLog(newValue);
+                }
+            },
+        },
         computed: {
             State() {
                 return State
             },
+            temporalLogs() {
+                if (!this.logs?.length) return [];
+
+                const filtered = this.logs.filter(log => {
+                    if (!this.filter) return true;
+                    return log.message?.toLowerCase().includes(this.filter.toLowerCase());
+                });
+
+                return filtered.map((logLine, index) => ({
+                    ...logLine,
+                    index,
+                    uniqueId: `${logLine.timestamp || ""}-${logLine.taskId || ""}-${index}`
+                }));
+            },
+            temporalLogIndicesByLevel()
```

Comment:
This property is used to group logs indices by level {
     "ERROR": ["1"],
     "WARN": [],
     "INFO": ["0", "2"],
     "DEBUG": [],
     "TRACE": []
}